README.md

```diff
@@ -37,6 +37,209 @@
+
+### 2.  Adopting the protocol.
+
+In your Class declaration, you *adopt* the protocol. Right now, the class declaration looks like this
+
+		class Person {
+
+Add the protocol so that it reads:
+
+		class Person: TextDescription
+		
+
+### 3. Conforming to the protocol 
+Conforming to a protocol means writing the code that implements it. In this case, you already have the code:
+
+In the Person class:
+
+		var textDescription: String {
+        		return "Name: \(self.name) \nHeight: \(self.height) \nFavorite Food: \(self.favoriteFood)"
+    		}
+
+For the Building class:
+		var textDescription: String {
+        		return "Address: \(self.address) \nHeight: \(self.height)"
+    		}
+    		
+This means that when you put the class declaration the protocol adoption, and the protocol conformance (implementation) together, here is what you get for Person:
+
+	class Person: TextDescription {
+
+    		// here are the properties
+    		var name: String
+    		var height: Double
+    		var favoriteFood: String
+
+		// this computed property implements the TextDescription protocol
+		var textDescription: String {
+			return "Name: \(self.name) \nHeight: \(self.height) \nFavorite Food: \(self.favoriteFood)"
+		}
+
+    		// here is the initializer method
+    		init(name: String, height: Double, favoriteFood: String) {
+        		self.name = name
+        		self.height = height
+        		self.favoriteFood = favoriteFood
+    		}
+	}
+
+Here is what you get for Building when the TextDescription protocol is adopted and implemented:
+		class Building: TextDescription {
+
+    			// here are the properties
+    			var address: String
+    			var height: Double
+			var textDescription: String {
+        			return "Address: \(self.address) \nHeight: \(self.height)"
+    			}
+
+    			// here is the initializer method
+    			init(address: String, height: Double) {
+        			address = name
+        			self.height = height
+   			}
+		}
+
+
+Declaring and using protocols means that you are structuring your code a bit more, and every time you structure code, you make it more robust. For example, if you adopt a protocol in your class declaration and don’t implement it, you’ll get a compiler error. 
+
+Also, you won’t wind up with a variety of properties such as `buildingTextDescription`, `personData`, and the like that all do the same thing. Every class that conforms to `TextClassDescriptio`n will have a property called `textDescription`. It’s your responsibility to make certain that it provides appropriate information, but you won’t be able to get mixed up and provide a `textDescription` for Person that uses address (that’s because address isn’t a property of Person and you can’t access it — you’ll get a compiler error).
+
```

Comment:
I think it would nice to show them this compile error (screenshot) after showing them a simple code snippet.

Like so:

Here we're declaring a protocol named `FavoriteIceCream`. It's one requirement (if you conform to it) is that you create a stored property (that is a variable) named `favoriteIceCream` of type `String. We are then creating a `Person` class which conforms to this `FavoriteIceCream` protocol.. but we don't have a stored property named `favoriteIceCream` on our Person class.

```swift
protocol FavoriteIceCream {
    
    var favoriteIcecream: String { get set }
    
}

class Person: FavoriteIceCream {
    
    var name: String
    
    init(name: String) {
        self.name = name
    }
    
}
```

Because of not adopting to the protocol (providing a `favoriteIceCream` property), we are met with the following error from the compiler:

![error](http://i.imgur.com/I8rjaXw.png)